File: cpp/easy/n_repeated_elements_in_size_2n_array.cpp
#include <unordered_set>
#include <vector>

int repeatedNTimes(std::vector<int>& nums) {
    
    std::unordered_set<int> seen;
    
    for (int num : nums) {
        if (seen.find(num) != seen.end()) {
            return num;
        }
        seen.insert(num);
    }

    return -1;
}

File: rust/easy/n_repeated_elements_in_size_2n_array.rs
use std::collections::HashSet;

pub fn repeated_n_times(nums: Vec<i32>) -> i32 {
    
    let mut seen = HashSet::new();
    
    for &number in &nums {
        if !seen.insert(number) {
            return number;
        }
    }
    
    -1
}